### PR TITLE
Remove Session Storage

### PR DESCRIPTION
**Pull request by mkevinosullivan**
### WHY are these changes introduced?

Supporting a variety of `SessionStorage` options and related dependencies causes stuff to break when those dependencies aren't supported on a particular platform (e.g., some cloud platforms).

Also, the responsibility for storing sessions is arguably more appropriate in the app.

### WHAT is this pull request doing?

- Removes the `session-storage` directory and updates CI/yarn scripts accordingly
- Where library functionality referenced the session storage to find a session, those method calls have been updated to take a session as an argument
- Update functionality to provide an app with a session id to search its storage for a session
- Make functionality available on the `Session` object to aid storage (`.serialize`)

## Type of change

- [x] Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have added/updated tests for this change
- [x] I have documented new APIs/updated the documentation for modified APIs (for public APIs)

### CODE REVIEW

File: docs/migrating-to-v6.md [hunk 1]
```diff
@@ -170,41 +170,7 @@ app.post('/webhooks', async (req, res) => {
...
-   ```
-
-   </div><div>After
-
-   ```ts
-   import {MemorySessionStorage} from '@shopify/shopify-api/session-storage/memory';
-   ```
-
-   </div>
+1. The `SessionStorage` interface has been removed and any provided implementions have been removed from the library.  The library only provides methods to obtain sessionId's.  Responsibility for storing sessions is delegated to the application.  Note: the previous implementations of session storage have been added to the `shopify-app-express` package, so that Express applications that use that package do not need to provide their own session storage.
```
[the remark this answers is not in the document]
Comment (mkevinosullivan) [reply docs/migrating-to-v6.md:173]: Yeah, there's a markdown file somewhere that I put a "TODO" in order to document this some more

File: docs/migrating-to-v6.md [hunk 2]
```diff
@@ -251,19 +217,36 @@ app.post('/webhooks', async (req, res) => {
...
+
+   ```ts
+   const callbackResponse = shopify.auth.callback({
+     isOnline: true,
+     rawRequest: req,
+     rawResponse: res,
+   });
+
+   // app stores Session in its own storage mechanism
+   await addSessionToStorage(callbackResponse.session.serialize());
```
[the remark this answers is not in the document]
Comment (mkevinosullivan) [reply docs/migrating-to-v6.md:247]: Would a separate `stringify()` method be more idiomatic, that would wrap the `serialize()`?

And have a corresponding `parse()` to put the string back into the object?

[the remark this answers is not in the document]
Comment (deldrid1) [reply docs/migrating-to-v6.md:247]: I would vote to leave it as JSON - apps can serialize on their own into binary if they like.  Debugging OAuth issues is a lot easier if you can look at various bits in plain text.

**Comment by deldrid1** [comment]
Any chance some of the session utilities can be preserved / maintained in this repo and exported per https://github.com/Shopify/shopify-api-node/issues/562?

File: docs/migrating-to-v6.md [hunk 3]
```diff
@@ -608,26 +592,9 @@ Here are all the specific changes that we made to the `Utils` object:
...
 
-1. `Shopify.Utils.deleteCurrentSession` is now `shopify.session.deleteCurrent`, it takes in an object, and the `isOnline` param is mandatory.
-   <div>Before
-
-   ```ts
-   const session = await Shopify.Utils.deleteCurrentSession(req, res);
-   ```
-
-   </div><div>:warning: After
+1. `Shopify.Utils.deleteCurrentSession` has been removed, as the library no longer handles the storage of sessions.
```
Comment (deldrid1) [inline docs/migrating-to-v6.md:595]: What is the best practice / recommendation for when an app should delete a session?

File: docs/migrating-to-v6.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mkevinosullivan) [reply docs/migrating-to-v6.md:247]: Agreed, let's just delegate the conversion to whatever storage format the app dev would prefer.

File: docs/migrating-to-v6.md (hunk 3 above)
Comment (mkevinosullivan) [reply docs/migrating-to-v6.md:595]: The main event to trigger deletion would be when the app is uninstalled by the shop (e.g., in response to `APP_UNINSTALLED` webhook).

File: docs/migrating-to-v6.md [hunk 4]
```diff
@@ -514,7 +497,8 @@ We also felt that the `Utils` object had some functions that belong to other par
 Here are all the specific changes that we made to the `Utils` object:
 
 1. `Shopify.Utils.generateLocalHmac` was removed because it's only meant to be used internally by the library.
-1. The `storeSession` method was removed since sessions shouldn't be stored using the library unless the library is doing it. Apps can still save data to their sessions as they please, as long as the data is properly exported to the library via the configured `SessionStorage`.
+1. The `storeSession` method was removed since sessions are no longer stored by the library. Apps are now fully responsible for implementating session storage and can save data to their sessions as they please ... TODO.
```
[the remark this answers is not in the document]
Comment (mkevinosullivan) [reply docs/migrating-to-v6.md:500]: Yes, working on it!

File: docs/migrating-to-v6.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkevinosullivan) [reply docs/migrating-to-v6.md:173]: https://www.npmjs.com/package/@shopify/shopify-app-express

File: docs/migrating-to-v6.md [hunk 5]
```diff
@@ -640,11 +640,11 @@ Here are all the specific changes that we made to the `Utils` object:
...
    const session = await getSessionFromStorage(sessionId);
 
-   const gqlClient = await shopify.clients.Graphql(session);
+   const gqlClient = await shopify.clients.Graphql({session});
    // or
-   const restClient = await shopify.clients.Rest(session);
+   const restClient = await shopify.clients.Rest({session});
    // or
-   const storefrontClient = await shopify.clients.Storefront(session);
+   const storefrontClient = await shopify.clients.Storefront({session});
```
Comment (paulomarg) [inline docs/migrating-to-v6.md:647]: There still seem to be some calls in this file that are using the old params for the clients, we should fix those.

**Review by paulomarg — APPROVED** [review]
🚢

**Pull request merged**